gems/aws-sdk-core/lib/aws-sdk-core/plugins/regional_endpoint.rb

```diff
@@ -47,14 +47,26 @@ class RegionalEndpoint < Seahorse::Client::Plugin
       # Legacy endpoints must continue to be generated at client time.
       option(:regional_endpoint, false)
 
-      # NOTE: All of the defaults block code is effectively deprecated.
-      # Because old services can depend on this new core version, we must
-      # retain it.
+      option(:ignore_configured_endpoint_urls, default: nil, doc_type: 'Boolean', docstring: <<-DOCS)
+Setting to true disables use of endpoint URLs provided via environment 
+variables and the shared configuration file.
+DOCS
+
+      # NOTE: with Endpoints 2.0, some of this logic is deprecated
+      # but because new old service gems may depend on new core versions
+      # we must preserve that behavior.
+      # Additional behavior controls the setting of the custom SDK::Endpoint
+      # parameter.
+      # When the `regional_endpoint` config is set to true - this indicates to
+      # Endpoints2.0 that a custom endpoint has NOT been configured by the user.
       option(:endpoint, doc_type: String, docstring: <<-DOCS) do |cfg|
 The client endpoint is normally constructed from the `:region`
 option. You should only configure an `:endpoint` when connecting
 to test or custom endpoints. This should be a valid HTTP(S) URI.
         DOCS
+
+        config_endpoint = resolve_custom_config_endpoint(cfg)
```

Comment:
config_endpoint is a confusing name for logic inside the :endpoint config. I'm not sure what better to name it. Does it make sense to refactor this entire block into multiple methods? Such as a `self.resolve_endpoint(cfg)` method and have that broken down into steps? Probably would be nice to do in a way that allows us to quickly identify (and shim/remove) deprecated bits.